src/common/Widgets/VpnMenuItem.vala

```diff
@@ -110,4 +113,31 @@ public class Network.VpnMenuItem : Gtk.ListBoxRow {
         w.no_show_all = !w.visible;
         w.hide ();
     }
+
+    private async void nap (uint interval, int priority = GLib.Priority.DEFAULT) {
+      GLib.Timeout.add (interval, () => {
+          nap.callback ();
+          return false;
+        }, priority);
+        yield;
+    }
+
+    /**
+    * Uses a timeout to check VPN connectivity
+    **/
+    private async void check_vpn_connectivity () {
+        if (!checking_vpn_connectivity) {
+
+            checking_vpn_connectivity = true;
+
+            for (int i = 0; i < 20; i++) {
```

Comment:
One rather subtle suggestion that I have, for future developers to make more sense of this, perhaps change the loop:

```for (int i = 0; i < 20; i++)``` 

I'd suggest replacing ```20``` with a more meaningful variable name. 
Perhaps make it a private static property on this class?